release/dist/qnap/pkgs.go

```diff
@@ -148,11 +110,103 @@ type qnapBuilds struct {
 	// It is used to avoid repeated compilations for the same architecture.
 	innerPkgs     dist.Memoize[*innerPkg]
 	dockerImageMu sync.Mutex
+	// tmpDir is a temp directory used for building qpkgs.
+	// It gets cleaned up when the dist.Build is closed.
+	tmpDir string
 }
 
 // getQnapBuilds returns the qnapBuilds for b, creating one if needed.
-func getQnapBuilds(b *dist.Build) *qnapBuilds {
-	return b.Extra(qnapBuildsMemoizeKey{}, func() any { return new(qnapBuilds) }).(*qnapBuilds)
+func getQnapBuilds(b *dist.Build, signer *signer) *qnapBuilds {
+	return b.Extra(qnapBuildsMemoizeKey{}, func() any {
+		builds := new(qnapBuilds)
+		if err := builds.setUpTmpDir(b, signer); err != nil {
+			panic(fmt.Errorf("setUpTmpDir: %v", err))
+		}
+		return builds
+	}).(*qnapBuilds)
+}
+
+//go:embed all:files
+var buildFiles embed.FS
+
+// setUpTmpDir copies the contents of the buildFiles embedded FS into the local
+// m.tmpDir for building.
+//
+// We do this to allow for this tailscale.com/release/dist/qnap package to be
+// used from both the corp and OSS repos. When built from OSS source directly,
+// this is a superfluous extra step, but when imported as a go module to another
+// repo (such as corp), we must do this to allow for the module's build files
+// to be reachable and editable from docker.
+//
+// This runs only once per dist.Build instance, is shared by all qnap targets,
+// and gets cleaned up upon close of the dist.Build.
+//
+// When a signer is provided, setUpTmpDir also sets up the qpkg signature files
+// in qbuild's expected location within m.tmpDir.
+func (m *qnapBuilds) setUpTmpDir(b *dist.Build, signer *signer) error {
```

Comment:
optional: this can be `func newQNAPBuilds(b *dist.Build, signed *signer) (*qnapBuilds, error)`

Reply:
Oh, good idea, I like that!